x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/machine_learning_alert_suppression.ts

```diff
@@ -0,0 +1,1082 @@
+              [ALERT_SUPPRESSION_START]: firstTimestamp,
+              [ALERT_SUPPRESSION_END]: firstTimestamp,
+              [ALERT_ORIGINAL_TIME]: firstTimestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 0,
+            })
+          );
+
+          const secondTimestamp = new Date().toISOString();
+          const secondAnomaly = {
+            ...baseAnomaly,
+            timestamp: secondTimestamp,
+          };
+
+          // Add more anomalies, then disable and re-enable to trigger another
+          // rule run. The second anomaly should trigger an update to the
+          // existing alert without changing the timestamp
+          await indexListOfDocuments([secondAnomaly, secondAnomaly]);
+          await patchRule(supertest, log, { id: createdRule.id, enabled: false });
+          await patchRule(supertest, log, { id: createdRule.id, enabled: true });
+          const secondAlerts = await getOpenAlerts(
+            supertest,
+            log,
+            es,
+            createdRule,
+            RuleExecutionStatusEnum.succeeded,
+            undefined,
+            new Date()
+          );
+
+          expect(secondAlerts.hits.hits).toHaveLength(1);
+          expect(secondAlerts.hits.hits[0]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'user.name',
+                  value: ['root'],
+                },
+              ],
+              [ALERT_ORIGINAL_TIME]: firstTimestamp,
+              [ALERT_SUPPRESSION_START]: firstTimestamp,
+              [ALERT_SUPPRESSION_END]: secondTimestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 2, // both new anomalies were suppressed into the original
+            })
+          );
+        });
+
+        it('suppresses alerts with array field values', async () => {
+          const timestamp = new Date().toISOString();
+          const anomaly = {
+            ...baseAnomaly,
+            'user.name': ['host1', 'host2'],
+            timestamp,
+          };
+          await indexListOfDocuments([anomaly, anomaly]);
+
+          const createdRule = await createRule(supertest, log, {
+            ...ruleProps,
+            from: timestamp,
+          });
+
+          const alerts = await getAlerts(supertest, log, es, createdRule);
+          expect(alerts.hits.hits).toHaveLength(1);
+          expect(alerts.hits.hits[0]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'user.name',
+                  value: ['host1', 'host2'],
+                },
+              ],
+              [ALERT_SUPPRESSION_START]: timestamp,
+              [ALERT_SUPPRESSION_END]: timestamp,
+              [ALERT_ORIGINAL_TIME]: timestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 1,
+            })
+          );
+        });
+
+        describe('with exceptions', () => {
```

Comment:
can we add tests for alerts enrichment and asset criticality, similarly as for other rule types?